pkg/operator/controller.go

```diff
@@ -122,6 +124,26 @@ func (c *Controller) createOrUpdateResources(cr *imageregistryv1.Config) error {
 }
 
 func (c *Controller) sync() error {
+	proxyConfig, err := clusterconfig.GetClusterProxyConfig(c.listers.ProxyConfigs)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster proxy configuration: %v", err)
+	}
+	if len(proxyConfig.Spec.NoProxy) != 0 {
+		if err := os.Setenv("NO_PROXY", proxyConfig.Spec.NoProxy); err != nil {
+			return fmt.Errorf("unable to set NO_PROXY env var: %v", err)
+		}
+	}
+	if len(proxyConfig.Spec.HTTPProxy) != 0 {
+		if err := os.Setenv("HTTP_PROXY", proxyConfig.Spec.HTTPProxy); err != nil {
+			return fmt.Errorf("unable to set HTTP_PROXY env var: %v", err)
+		}
+	}
+	if len(proxyConfig.Spec.HTTPSProxy) != 0 {
+		if err := os.Setenv("HTTPS_PROXY", proxyConfig.Spec.HTTPSProxy); err != nil {
+			return fmt.Errorf("unable to set HTTPS_PROXY env var: %v", err)
+		}
+	}
```

Comment:
It won't reflect changes of the proxy config made after the operator is started.

Reply:
the operator will need to watch the global proxy config resource and terminate itself (so it gets restarted) if the config changes.

Reply:
Can we just reset these environment variables at the beginning of the reconciliation cycle? I'd like to avoid self-termination.

Reply:
yeah i guess there's technically no reason you have to actually restart the process, just updating the env vars should be sufficient as long as the aws sdk (or other client sdks you use) don't cache/store the proxy config somewhere on initialization.

Reply:
Couldn't I just remove the `len(proxyConfig.Spec.<proxyinfo>) != 0` and if the proxyConfig variables are empty they will be set as empty?  I believe that to be the same as resetting them, and then setting them again if the` len() != 0` is true, it just saves some code and is possibly more clear.

Reply:
At a minimum you need to ensure that this sync logic is being invoked any time the global proxy config resource (or the operator's own proxy config resource) changes.  I'm not sure when this sync() function is called currently.

Always setting the env var to the current value (whether it's emptystring or a real value) is probably fine, but you should test that behavior in go to ensure Setenv doesn't no-op if you pass an emptystring or something else weird.

and lastly, per my other comment, you need to ensure that any client tools you're using are re-reading the env vars every time you invoke them.  If they are caching this value, you'll need to find a way to reset the cache, or terminate the process entirely to force a re-init.

Reply:
Yes, removing `len() != 0` should be enough.

Reply:
afaik the sync process runs whenever an event is encountered by the listers.  So as long as the lister is watching the cluster proxy config, it will run.  @dmage, can you weight in on that?

Reply:
yes, right, we call `sync` on every change event from informers (exception is events for ClusterOperators).

Reply:
@bparees does not seem to throw an error on setting an empty string: https://play.golang.org/p/PUKIQ7NRLKj

Reply:
> Can we just reset these environment variables at the beginning of the reconciliation cycle? I'd like to avoid self-termination.

I don't like the idea of just setting the env vars inside of the pod, it seems very black box.  I think the deployment should get updated with the new proxy configuration (which shouldn't happen very often) and then the operator redeploys with the new settings.  This would allow an admin to query the deployment for the operator and see that a proxy is set and being used.

@dmage thoughts?

Reply:
@bparees can we ask CVO to set these environment variables for all Deployments?

Reply:
that was discussed briefly during some of the proxy design discussions but i don't think it's got a lot of support...  that's a pretty big hammer.  (And @coreydaley unless the CVO sets them, we can't do what you want)

/cc @danehans

Reply:
@bparees are there any other operators that you know of that have already implemented this proxy behavior that I can use as a reference? Or are we the guinea pigs yet again?

Reply:
i'm not familiar w/ any.